src/lib.rs

```diff
@@ -336,18 +340,66 @@ impl Config {
         assert!(output.ends_with(".a"));
         let lib_name = &output[3..output.len() - 2];
         let dst = self.get_out_dir();
+        let msvc = self.get_target().contains("msvc");
+        let compiler = self.get_compiler();
 
         let mut objects = Vec::new();
         let mut src_dst = Vec::new();
+        let mut dependencies = HashMap::new();
         for file in self.files.iter() {
             let obj = dst.join(file).with_extension("o");
             let obj = if !obj.starts_with(&dst) {
                 dst.join(obj.file_name().unwrap())
             } else {
                 obj
             };
+
+            let mut input_mtime = get_mtime(&file);
+            let output_mtime = get_mtime(&obj);
+            // If input file is older than output object, check its dependencies
+            if input_mtime < output_mtime {
+                let mut cmd = compiler.to_command();
+                let files: Vec<PathBuf>;
+                cmd.arg(&file);
+                if msvc {
+                    cmd.arg("/showIncludes").arg("/E");
+                    println!("running: {:?}", cmd);
+                    files = cmd.output()
```

Comment:
Yeah, the double `unwrap()` is offensive and embarrassing. I think I got way too caught up in the excitement of extreme chaining. I'll sort that out.